tests: fix mgd and dpd tcp port regex

rcgoodfellow opened the pull request:
The regular expressions for extracting `mgd` and `dpd` TCP ports from logs have issues that are fixed by this PR.

I considered short-circuiting the launching of `mgd` from the nexus test suite entirely, since it's not really being used due to #5780 (and that is why we did not catch this regex issue earlier). However, that would involve unwinding a fair amount of test infrastructure that we ultimately want to keep in place, we just need to come up with a solution for #5780. So I've left `mgd` in place for now and just fixed the regular expressions.

bnaecker commented:
Fixes for the regexes themselves look good to me. I think we should add a test or two for each of them though.

While we're here, we may also want to pull in a similar fix as that in https://github.com/oxidecomputer/omicron/pull/3640, specifically these lines:

https://github.com/oxidecomputer/omicron/blob/509bf0727e8bb98d88ed7346ce6469d70bd3c848/test-utils/src/dev/clickhouse.rs#L712-L715

That will ensure that, should we hit EOF partway through the needle the regex is searching for, we can still eventually recover it. That will catch an edge case that's pretty difficult to debug, because post-hoc the log files all look correct -- it's only when reading them as they're being written that the error appeared.

rcgoodfellow commented:
Thanks for the feedback @bnaecker. I've added a port discovery test for `mgd` based on the one for `dpd`. Also added the reader fix for both `dpd` and `mgd`.

bnaecker reviewed (approved):
Looks good, thanks for adding the test!
